contracts/OperatorStage2.sol

```diff
@@ -48,12 +65,33 @@ contract OperatorStage2 is AccessControl {
 		tokenGateway = StandardTokenGateway(gatewayAddress);
 	}
 
+	// Adds a new rate that allows a user to bond with
+	function addRate(uint256 _rate, uint256 _maturityInWeeks) public onlyOperatorStage2 {
+		allowedYieldToWeeks[_rate] = _maturityInWeeks;
+		BondRate memory br = BondRate(_rate, _maturityInWeeks);
+		ratesAvailable.push(br);
+		emit Yield(_rate, _maturityInWeeks);
+	}
+
+	// Sets a rate to zero, not removing it but making it obsolete
+	// TODO: make this nicer and not just bond for 0% yield
+	function removeRate(uint256 _rate) public onlyOperatorStage2 {
+		allowedYieldToWeeks[_rate] = 0;
```

Comment:
also we need to loop through `ratesAvailable` to delete from there, right?

Reply:
it's a way to delete an item to keep order, yes. for simplicity I've not opted for it.

Reply:
oh yeah fine with the method, i just meant we need to delete the rate from that array